File: csharp/LogicAndTrick.WikiCodeParser/Nodes/UnprocessablePlainTextNode.cs
using System;
using System.Collections.Generic;

namespace LogicAndTrick.WikiCodeParser.Nodes
{
    public class UnprocessablePlainTextNode : INode
    {
        public static INode Empty => new UnprocessablePlainTextNode(string.Empty);
        public static INode NewLine => new UnprocessablePlainTextNode("\n");

        public string Text { get; set; }

        public UnprocessablePlainTextNode(string text)
        {
            Text = text;
        }

        public string ToHtml() => HtmlHelper.Encode(Text);
        public string ToPlainText() => Text;
        public IList<INode> GetChildren() => Array.Empty<INode>();
        public void ReplaceChild(int i, INode node) => throw new InvalidOperationException();
        public bool HasContent() => !String.IsNullOrWhiteSpace(Text);
    }
}

File: php/src/Nodes/UnprocessablePlainTextNode.php
<?php

namespace LogicAndTrick\WikiCodeParser\Nodes;

use LogicAndTrick\WikiCodeParser\HtmlHelper;

class UnprocessablePlainTextNode implements INode
{
    public static function Empty(): INode
    {
        return new UnprocessablePlainTextNode('');
    }

    public static function NewLine(): INode
    {
        return new UnprocessablePlainTextNode("\n");
    }

    public string $text;

    public function __construct(string $text)
    {
        $this->text = $text;
    }

    function ToHtml(): string
    {
        return HtmlHelper::Encode($this->text);
    }

    function ToPlainText(): string
    {
        return $this->text;
    }

    function GetChildren(): array
    {
        return [];
    }

    function ReplaceChild(int $i, INode $node): void
    {
        throw new \Exception('Invalid operation');
    }

    function HasContent(): bool
    {
        return $this->text && trim($this->text) != '';
    }
}